src/simplifier.h

```diff
@@ -0,0 +1,23 @@
+
+#ifndef _SIMPLIFIER_H
```

Comment:
Let's call the file expression_pruner instead of simplifier. Simplifier is too generic.